src/SignalsHandler.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace React\EventLoop;
+
+use Evenement\EventEmitter;
+
+/**
+ * @internal
+ */
+final class SignalsHandler extends EventEmitter
+{
+    private $signals = [];
+
+    public function add($signal, callable $listener)
+    {
+        if (!isset($this->signals[$signal])) {
+            $this->signals[$signal] = [];
+
+            $this->emit('on', [$signal]);
+        }
+
+        $this->signals[$signal][] = $listener;
+    }
+
+    public function remove($signal, callable $listener)
+    {
+        if (!isset($this->signals[$signal])) {
+            return;
+        }
+
+        $index = \array_search($listener, $this->signals[$signal], true);
+        while (false !== $index) {
+            unset($this->signals[$signal][$index]);
+            $index = \array_search($listener, $this->signals[$signal], true);
+            if (isset($this->signals[$signal]) && \count($this->signals[$signal]) === 0) {
+                unset($this->signals[$signal]);
+                $this->emit('off', [$signal]);
+            }
```

Comment:
Yup, kinda depends on the result of the discussion here: https://github.com/reactphp/event-loop/pull/104#discussion_r132964188